snuba/writer.py

```diff
@@ -7,12 +7,16 @@
 
 
 class BatchWriter(object):
-    def write(self, schema, rows):
+    def __init__(self, schema):
+        raise NotImplementedError
+
+    def write(self, rows):
         raise NotImplementedError
 
 
 class NativeDriverBatchWriter(BatchWriter):
-    def __init__(self, connection):
+    def __init__(self, schema, connection):
+        self.__schema = schema
```

Comment:
This is one of those times that I really wish Python had real protected variables. :(